PWG/EMCAL/EMCALtasks/AliEmcalCorrectionComponent.cxx

```diff
@@ -208,9 +208,22 @@ void AliEmcalCorrectionComponent::UpdateCells()
   
   Int_t bunchCrossNo = fEventManager.InputEvent()->GetBunchCrossNumber();
   
-  if (fRecoUtils)
+  if (fRecoUtils){
+    //In case of PAR run check global event ID
+    Short_t currentParIndex = 0;
+    if(fRecoUtils->IsParRun()){
+      ULong64_t globalEventID = (ULong64_t)bunchCrossNo + (ULong64_t)fEventManager.InputEvent()->GetOrbitNumber() * (ULong64_t)3564 + (ULong64_t)fEventManager.InputEvent()->GetPeriodNumber() * (ULong64_t)59793994260;
+      for(Short_t ipar=0;ipar<fRecoUtils->GetNPars();ipar++){
+	if(globalEventID >= fRecoUtils->GetGlobalIDPar(ipar)) {
+	  currentParIndex++;
+	}
+      }
+    }
+    fRecoUtils->SetCurrentParNumber(currentParIndex);
```

Comment:
This was not here before so even if par is off why do we set it here?

Reply:
currentParIndex is set to 0. And 0 means that the regular calibration before any PAR is taken.

Reply:
In priniple it is safe to go with this setting into the condition statement.

Reply:
I would prefer the latter...